app/validators/checksum_validator.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class ChecksumValidator < ActiveModel::Validator
+  include Rails.application.routes.url_helpers
+  def validate(record)
+    @record = record
```

Comment:
@RudyOnRails I would make this an `attr_reader` and loose and just use `record` throughout for clarity. It seems more idiomatic to me.